server.py

```diff
@@ -386,6 +393,115 @@ async def create_cluster(  # pylint: disable=too-many-arguments,too-many-positio
     return cluster.id
 
 
+@mcp.tool()
+@track_tool_usage()
+async def validate_nmstate_yaml(nmstate_yaml: str) -> str:
+    """
+    Validate an nmstate yaml document.
+
+    The yaml should always be validated before submitted to the cluster.
+    """
+    validate_and_parse_nmstate(nmstate_yaml)
+    return "YAML is valid"
+
+
+@mcp.tool(
+    description=f"""
+    Generate an initial nmstate yaml.
+
+    You should call this after gathering information from the user to generate the initial nmstate
+    yaml. Then you can tweak it as needed. Do not generate nmstate yaml from scratch without calling
+    this tool.
+
+    Returns: the generated nmstate yaml
+
+    Input param schema:
+    {NMStateTemplateParams.model_json_schema()}
+"""
+)
+@track_tool_usage()
+async def generate_nmstate_yaml(params: NMStateTemplateParams) -> str:
+    """
+    Generate an initial nmstate yaml.
+
+    See the mcp.tool description for more details (we have to use it since we dynamically generate
+    the input schema for the params).
+    """
+    log.info("Generate nmstate yaml with params: %s", params.model_dump_json(indent=2))
+    try:
+        generated = generate_nmstate_from_template(params)
+        log.debug("Generated yaml: %s", generated)
+        return generated
+    except TemplateError as e:
+        log.error("Failed to render nmstate template", exc_info=e)
+        return "ERROR: Failed to generate nmstate yaml"
+    except Exception as e:
+        log.error("Exception generating nmstate yaml", exc_info=e)
+        return "ERROR: Unknown error"
+
+
+@mcp.tool()
+@track_tool_usage()
+async def alter_static_network_config_nmstate_for_host(
+    cluster_id: str, index: int | None, new_nmstate_yaml: str | None
+) -> str:
+    """
+    Add, replace, or delete the nmstate yaml for a single host.
+
+    Args:
+        cluster_id (str): The unique identifier of the cluster
+        index (int | None): The index of the host in the existing static config to replace, or None to append a new host to the end of the config.
```

Comment:
Is an index the best way to handle this? Are we sure the ordering is being preserved? I don't feel like there's a reason to preserve it on the assisted side so I'm a bit worried that even if this works now it might not be intentional or good to rely on.